### PR TITLE
Added route for wish details, cleaned up wish details and childInfo, …

### PR DESCRIPTION
**Pull request by aswin1689**
Added route for wish details, cleaned up wish details and childInfo, added unit tests

### What was the feature/problem?

No route from Wish summary page to go to Wish Details.

### How does this PR address the above feature/problem?

Add route for WishDetails page and provide a link to navigate between wish summary and wish details pages

### Check lists (check `x` in `[ ]` of list items)

- [ x] Test passed
- [x ] Coding style (indentation, etc)

### CODE REVIEW

File: ui/src/childinfo/ChildInfo.js [hunk 1]
```diff
@@ -62,25 +59,8 @@ export default class ChildInfo extends Component {
       });
     } else {
 
-      const { name, age, homeTown, illness, details } = this.state
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.js]: I think we still need to have the POST makeAWish call here in order to have an id to pass in to the WishDetails component

File: ui/src/childinfo/ChildInfo.js [hunk 2]
```diff
@@ -62,25 +59,8 @@ export default class ChildInfo extends Component {
...
-        },
-        type: type ? type : '',
-        details: details
-      }
-
-      const response = await WishDetailsService.makeAWish(wish)
-
       this.setState({
-        showConfirmation: true,
-        childId: response._id
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.js]: Think we need this response._id saved, same as above

File: ui/src/wishDetails/WishDetails.js [hunk 3]
```diff
@@ -100,7 +103,3 @@ export default class WishDetails extends Component {
     )
   }
 }
-
```
Comment (jrgoodpasture) [inline ui/src/wishDetails/WishDetails.js]: Should `id` be added here instead?

File: ui/src/childinfo/ChildInfo.js (hunk 1 above)
Comment (aswin1689) [reply ui/src/childinfo/ChildInfo.js]: added back that call

File: ui/src/childinfo/ChildInfo.js (hunk 2 above)
Comment (aswin1689) [reply ui/src/childinfo/ChildInfo.js]: added it back

File: ui/src/wishDetails/WishDetails.js (hunk 3 above)
Comment (aswin1689) [reply ui/src/wishDetails/WishDetails.js]: I am matching it from params instead. We will add proptypes for all components later

**Pull request merged**